src/apps/dashboard/routes/branding/index.tsx

```diff
@@ -107,16 +189,75 @@ export const Component = () => {
+                        )}
+
+                        <Stack
+                            direction={{
+                                xs: 'column',
+                                sm: 'row'
+                            }}
+                            spacing={3}
+                        >
+                            <Box sx={{ flex: '1 1 0' }}>
+                                <Image
+                                    isLoading={false}
+                                    url={
+                                        brandingOptions.SplashscreenEnabled ?
+                                            splashscreenUrl :
+                                            undefined
+                                    }
                                 />
-                            }
-                            label={globalize.translate('EnableSplashScreen')}
-                        />
+                            </Box>
+
+                            <Stack
+                                spacing={{ xs: 3, sm: 2 }}
+                                sx={{ flex: '1 1 0' }}
+                            >
+                                <FormControlLabel
+                                    control={
+                                        <Switch
+                                            name={BrandingOption.SplashscreenEnabled}
+                                            checked={brandingOptions?.SplashscreenEnabled}
+                                            onChange={setSplashscreenEnabled}
+                                        />
+                                    }
+                                    label={globalize.translate('EnableSplashScreen')}
+                                />
+
+                                <Typography variant='body2'>
+                                    {globalize.translate('CustomSplashScreenSize')}
+                                </Typography>
+
+                                <Button
+                                    component='label'
+                                    variant='outlined'
+                                    startIcon={<Upload />}
+                                    disabled={!brandingOptions.SplashscreenEnabled}
+                                >
+                                    <input
+                                        type='file'
+                                        accept='image/*'
+                                        hidden
+                                        onChange={onSplashscreenUpload}
+                                    />
+                                    {globalize.translate('UploadCustomImage')}
+                                </Button>
+
+                                <Button
+                                    variant='outlined'
+                                    color='error'
+                                    startIcon={<Delete />}
+                                    disabled={!brandingOptions.SplashscreenEnabled}
```

Comment:
Would it be possible to disable this button if there is no custom splash screen set? Or do we not have enough information to know that?

Reply:
As far as I know we have no way of knowing that.